src/main/java/com/documaster/rms/noark/ws/client/handlers/ErrorResponseHandler.java

```diff
@@ -1,17 +1,46 @@
 package com.documaster.rms.noark.ws.client.handlers;
 
 import java.io.IOException;
+import java.io.InputStream;
 import java.nio.charset.StandardCharsets;
 
+import com.documaster.rms.noark.ws.client.exceptions.BadRequestException;
 import com.documaster.rms.noark.ws.client.exceptions.ConflictException;
 import com.documaster.rms.noark.ws.client.exceptions.InsufficientPrivilegesException;
+import com.documaster.rms.noark.ws.client.exceptions.ResourceNotFoundException;
 import com.documaster.rms.noark.ws.client.exceptions.ServiceException;
 import com.documaster.rms.noark.ws.client.exceptions.UnauthorizedException;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import org.apache.commons.io.IOUtils;
 import org.apache.http.HttpResponse;
 import org.apache.http.HttpStatus;
 
-class ErrorResponseHandler {
+public class ErrorResponseHandler {
+
+	private final ObjectMapper mapper;
+	private final boolean isApplicationJsonResponseTypeEnabled;
+
+	public ErrorResponseHandler() {
+
+		this(null);
+	}
+
+	public ErrorResponseHandler(ObjectMapper mapper) {
+
+		this(mapper, false);
+	}
+
+	public ErrorResponseHandler(ObjectMapper mapper, boolean isApplicationJsonResponseTypeEnabled) {
```

Comment:
Do you think it would be better if the `ErrorResponseType` was passed instead of the `isApplicationJsonResponseTypeEnabled` flag? Then the `ErrorResponseHandler` would decide weather or not to construct an `ErrorMessage`.

I don't like the current approach as the logic for evaluating the flag is duplicated, and if we added another response type, a second flag (?) would have to be added to the constructor.